From 8fa3f8e368baaf8958dacd5e519dcabe41a374c7 Mon Sep 17 00:00:00 2001
From: jiajia Qian <jiajia.qian@nxp.com>
Date: Fri, 22 Mar 2024 13:32:48 +0800
Subject: [PATCH] tests: Fix cl test api@clenqueuemigratememobjects crash.
Upstream-Status: Pending [https://gitlab.freedesktop.org/mesa/piglit/-/merge_requests/892]

clEnqueueMigrateMemObjects is async, calling clReleaseMemObject before it completes will cause segment fault.

So add a clfinish function after clEnqueueMigrateMemObjects could fix the issue, no more segment fault.

Signed-off-by: jiajia Qian <jiajia.qian@nxp.com>
---
 tests/cl/api/enqueue-migrate-mem-objects.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/cl/api/enqueue-migrate-mem-objects.c b/tests/cl/api/enqueue-migrate-mem-objects.c
index f964360b3..14ec163c0 100644
--- a/tests/cl/api/enqueue-migrate-mem-objects.c
+++ b/tests/cl/api/enqueue-migrate-mem-objects.c
@@ -236,6 +236,7 @@ piglit_cl_test(const int argc,
 	     "CL_INVALID_EVENT_WAIT_LIST if event objects in event_wait_list are not valid events");
 
 
+	clFinish(queue);
 	clReleaseMemObject(device_buffer);
 	return result;
 #else
-- 
2.34.1

